### PR TITLE
Added method to load from "json file"

### PR DESCRIPTION
**Pull request by LightOri**
- added method to load from "json file";
- changed data output for jQuery and XHR

### CODE REVIEW

File: scripts/JSON.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const jsonButton = document.getElementById('json').addEventListener("click", getData);
+
+function getData() {
+  var xhr = new XMLHttpRequest();
```
Comment (ghost) [inline scripts/JSON.js:4]: use **let** instead

File: scripts/JSON.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  xhr.open( 'GET', 'data/data.json', true );
+  xhr.onreadystatechange = function() {
+   if( this.readyState === 4 && this.status !== 404 ) {
```
Comment (ghost) [inline scripts/JSON.js:7]: move 4 and 404 to consts.

File: scripts/JSON.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    jsonHandler( this.responseText );
```
Comment (ghost) [inline scripts/JSON.js:8]: it is private method

File: scripts/JSON.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+   }
+  }
+  xhr.send();
+}
+
+function jsonHandler(data) {
```
Comment (ghost) [inline scripts/JSON.js:14]: incapsulate it in "getData function"

File: scripts/JSON.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	const resultsContainer = document.getElementById('result');
+	const dataList = JSON.parse(data);
+
```
Comment (ghost) [inline scripts/JSON.js:17]: remove empty line

File: scripts/JSON.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  const colorsArray = dataList.colorsArray
+  const colorsList = colorsArray.map(key => '<br>' + key.colorName.toUpperCase() + ' (' + key.hexValue + ')')
```
Comment (ghost) [inline scripts/JSON.js:19]: - align consts,
- use template string
- show colorNames with appropriate colors (inline styles for example)

File: scripts/Promises.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+const promiseButton = document.getElementById('promise').addEventListener("click", getData);
```
Comment (ghost) [inline scripts/Promises.js:1]: Don't commit WIP code

File: scripts/jQueryAJAX.js [hunk 8]
```diff
@@ -12,6 +12,19 @@ $('#jQueryAJAX').click(function(){
 
 function jQueryAJAXHandler(data) {
 	const resultsContainer = document.getElementById('result');
-	resultsContainer.innerHTML = JSON.stringify(data);
+
+	//user data
+	const email = data.data.profile.email
+	const firstname = data.data.profile.firstname
+	const lastname = data.data.profile.lastname
+	const company = data.data.profile.company
```
Comment (ghost) [inline scripts/jQueryAJAX.js:20]: `const profile = data.data.profile;
const email = profile.email
const firstname = profile.firstname;
const lastname = profile.lastname;
const company = profile.company;`

File: scripts/jQueryAJAX.js [hunk 9]
```diff
@@ -12,6 +12,19 @@ $('#jQueryAJAX').click(function(){
...
 	
-};
\ No newline at end of file
+	const userData = 'Firstname: ' + firstname + '<br>'
```
Comment (ghost) [inline scripts/jQueryAJAX.js:22]: use template string

File: scripts/xmlHttpRequest.js [hunk 10]
```diff
@@ -22,6 +22,21 @@ function mockXhrData(){
 
 function xhrHandler(data) {
 	const resultsContainer = document.getElementById('result');
-	resultsContainer.innerHTML = data;
+	const parsedData = JSON.parse(data);
+
+	//user data
+	const email = parsedData.data.profile.email
```
Comment (ghost) [inline scripts/xmlHttpRequest.js:28]: the same with profile as in Jquery Ajax method

File: scripts/JSON.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
+const jsonButton = document.getElementById('json').addEventListener("click", getData);
```
Comment (ghost) [inline scripts/JSON.js:1]: encapsulate every file with IIFE

File: styles/style.css [hunk 12]
```diff
@@ -13,6 +13,7 @@ aside li:first-child {
 	text-align: center;
 }
 
+aside li:nth-child(2),
 aside li:nth-child(3),
 aside li:nth-child(4) {
```
Comment (ghost) [inline styles/style.css:18]: Find another way to add styles for 2-4 LIs

**Review by ghost — CHANGES_REQUESTED** [review]

File: scripts/JSON.js (hunk 11 above)
Comment (LightOri) [reply scripts/JSON.js:1]: fixed

File: scripts/JSON.js (hunk 1 above)
Comment (LightOri) [reply scripts/JSON.js:4]: fixed

File: scripts/JSON.js (hunk 2 above)
Comment (LightOri) [reply scripts/JSON.js:7]: fixed

File: scripts/JSON.js (hunk 3 above)
Comment (LightOri) [reply scripts/JSON.js:8]: need some time for investigating this issue. hope, will fix it soon

File: scripts/JSON.js (hunk 4 above)
Comment (LightOri) [reply scripts/JSON.js:14]: need some time for investigating this issue. hope, will fix it soon

File: scripts/JSON.js (hunk 5 above)
Comment (LightOri) [reply scripts/JSON.js:17]: fixed

File: scripts/JSON.js (hunk 6 above)
Comment (LightOri) [reply scripts/JSON.js:19]: fixed

File: scripts/jQueryAJAX.js (hunk 8 above)
Comment (LightOri) [reply scripts/jQueryAJAX.js:20]: fixed

File: scripts/Promises.js (hunk 7 above)
Comment (LightOri) [reply scripts/Promises.js:1]: ok, thx!

File: scripts/jQueryAJAX.js (hunk 9 above)
Comment (LightOri) [reply scripts/jQueryAJAX.js:22]: fixed

File: scripts/xmlHttpRequest.js (hunk 10 above)
Comment (LightOri) [reply scripts/xmlHttpRequest.js:28]: fixed

File: styles/style.css (hunk 12 above)
Comment (LightOri) [reply styles/style.css:18]: already fixed in DDT-1 branch

**Comment by LightOri** [comment]
10 of 12 issues were fixed
2 other will be fixed soon

**Pull request closed without merging**